src/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,119 @@
+import { List } from 'immutable';
+import truncate from 'lodash.truncate';
+
+import { SxProps } from '@mui/material';
+import Breadcrumbs from '@mui/material/Breadcrumbs';
+import Typography from '@mui/material/Typography';
+import { styled } from '@mui/material/styles';
+
+import React from 'react';
+
+import { ItemType } from '@graasp/sdk';
+import { ItemRecord } from '@graasp/sdk/frontend';
+
+import ItemMenu, { ItemMenuProps } from './ItemMenu';
+import { CenterAlignWrapper, ITEM_NAME_MAX_LENGTH, StyledLink } from './utils';
+
+const StyledBreadcrumbs = styled(Breadcrumbs)(({ theme }) => ({
+  '& ol': {
+    textIndent: -theme.typography.fontSize,
+  },
+}));
+
+export type NavigationProps = {
+  backgroundColor?: string;
+  buildBreadcrumbsItemLinkId?: (id: string) => string;
+  buildIconId?: (id: string) => string;
+  buildMenuItemId?: (id: string) => string;
+  buildToItemPath: (id: string) => string;
+  buildMenuId?: (id: string) => string;
+  id?: string;
+  item?: ItemRecord;
+  parents?: List<ItemRecord>;
+  renderRoot?: (item?: ItemRecord) => JSX.Element | null;
+  sx?: SxProps;
+  useChildren: ItemMenuProps['useChildren'];
+  maxItems?: number;
+};
+
+const Navigation = ({
+  backgroundColor,
+  buildBreadcrumbsItemLinkId,
+  buildIconId,
+  buildMenuItemId,
+  buildToItemPath,
+  id,
+  item,
+  parents,
+  renderRoot,
+  sx,
+  useChildren,
+  buildMenuId,
+  maxItems = 4,
+}: NavigationProps): JSX.Element | null => {
+  const renderParents = (): JSX.Element[] | undefined =>
+    parents?.toJS()?.map(({ name, id }) => (
```

Comment:
Is the conversion to JS necessary here ?

Reply:
Yes because otherwise it returns a `List<Element>`, but I've changed to `toArray` so it's a bit more explicit.